src/Grpc.Net.Client/Balancer/Internal/ConnectionManager.cs

```diff
@@ -148,30 +151,65 @@ private void OnResolverResult(ResolverResult result)
                 throw new InvalidOperationException($"Load balancer not configured.");
             }
 
+            var channelStatus = result.Status;
+
             // https://github.com/grpc/proposal/blob/master/A21-service-config-error-handling.md
-            // 1. Only use resolved service config if not disabled.
-            // 2. Only use resolved service config if valid. This includes having a factory for one of the load balancing configs.
+            // Additionally, only use resolved service config if not disabled.
             LoadBalancingConfig? loadBalancingConfig = null;
-            if (result.ServiceConfig != null)
+            if (!DisableResolverServiceConfig)
             {
-                if (!DisableResolverServiceConfig)
+                ServiceConfig? workingServiceConfig = null;
+                if (result.ServiceConfig == null)
                 {
-                    if (result.ServiceConfig.LoadBalancingConfigs.Count > 0)
+                    // Step 4 and 5
+                    if (result.ServiceConfigStatus == null)
+                    {
+                        // Step 5
+                        _previousServiceConfig = DefaultServiceConfig;
+                        workingServiceConfig = DefaultServiceConfig;
+                    }
+                    else
                     {
-                        if (!ChildHandlerLoadBalancer.TryGetValidServiceConfigFactory(result.ServiceConfig.LoadBalancingConfigs, LoadBalancerFactories, out loadBalancingConfig, out var _))
+                        // Step 4
+                        if (_previousServiceConfig == null)
                         {
-                            ConnectionManagerLog.ResolverUnsupportedLoadBalancingConfig(Logger, result.ServiceConfig.LoadBalancingConfigs);
+                            // Step 4.ii
```

Comment:
```suggestion
                            // Step 4.ii: If no config was provided or set previously, then treat resolution as a failure.
```

BTW: I saw the spec said that the channel should be in a `TRANSIENT_FAILURE` state at this point. Where does that happen?